File: ts/src/algorithms/mixed/reverse-vowels/reverse.test.ts
import {test, expect} from "bun:test";
import {reverseVowels} from "./reverse";

test("reverseVowels", () => {
  let res = reverseVowels("hello");
  expect(res).toBe("holle");
});

test("reverseVowels", () => {
  let res = reverseVowels("leetcode");
  expect(res).toBe("leotcede");
});

test("reverseVowels", () => {
  let res = reverseVowels("aA");
  expect(res).toBe("Aa");
});

File: go/src/algorithms/mixed/reverse-vowels/reverse_test.go
package mixed

import "testing"

func TestReverseVowels(t *testing.T) {
	testCases := []struct {
		input    string
		expected string
	}{
		{"aA", "Aa"},
		{"leetcode", "leotcede"},
		{"hello", "holle"},
	}

	for _, tc := range testCases {
		output := reverseVowels(tc.input)
		if output != tc.expected {
			t.Errorf("Test failed for input %v, expected: '%v', got:  '%v'", tc.input, tc.expected, output)
		}
	}
}
